intake/catalog/base.py

```diff
@@ -259,6 +331,36 @@ def __getattr__(self, item):
                 raise AttributeError(item)
         raise AttributeError(item)
 
+    def __setitem__(self, name, entry):
+        """Add entry to catalog
+
+        This relies on the `_entries` attribute being mutable, which it normally
+        is. Note that if a catalog automatically reloads, any entry added here
+        may be very transient
+
+        Parameters
+        ----------
+        name : str
+            Key to give the entry in the cat
+        entry : CatalogEntry
+            The entry to include (could be local, remote)
+        """
+        self._entries[name] = entry
+
+    def __delitem__(self, name):
+        """Remove entry from catalog
+
+        This relies on the `_entries` attribute being mutable, which it normally
+        is. Note that if a catalog automatically reloads, any entry removed here
+        may soon reappear
+
+        Parameters
+        ----------
+        name : str
```

Comment:
I think we need to work hard to differentiate between key and name. Also seems like it would be safer to use the value rather than the name for this also I mildly prefer `remove`. Maybe `pop` could be used for key?